pkg/generators/git.go

```diff
@@ -1,25 +1,101 @@
 package generators
 
 import (
+	"context"
 	"fmt"
 	argoprojiov1alpha1 "github.com/argoproj-labs/applicationset/api/v1alpha1"
+	"github.com/argoproj-labs/applicationset/pkg/utils"
 	argov1alpha1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/reposerver/apiclient"
+	log "github.com/sirupsen/logrus"
+	"path"
 )
 
 var _ Generator = (*GitGenerator)(nil)
 
 type GitGenerator struct {
+	repoClientset apiclient.Clientset
 }
 
-func NewGitGenerator() Generator {
-	g := &GitGenerator{}
+func NewGitGenerator(repoServerAddress string) Generator {
+	repoClientset := apiclient.NewRepoServerClientset(repoServerAddress, 5)
+	g := &GitGenerator{
+		repoClientset: repoClientset,
+	}
 	return g
 }
 
-func (g *GitGenerator) GenerateApplications(appSet *argoprojiov1alpha1.ApplicationSet) ([]argov1alpha1.Application, error) {
-	if appSet == nil {
+func (g *GitGenerator) GenerateApplications(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator, appSet *argoprojiov1alpha1.ApplicationSet) ([]argov1alpha1.Application, error) {
+	if appSetGenerator == nil || appSet == nil {
 		return nil, fmt.Errorf("ApplicationSet is empty")
 	}
 
+	if appSetGenerator.Git == nil {
+		return nil, fmt.Errorf("git variable empty")
+	}
+
+	if len(appSetGenerator.Git.Directories) > 0 {
+		paths, err := g.GetAllPaths(appSetGenerator.Git)
+		if err != nil {
+			return nil, err
+		}
+
+		res := make([]argov1alpha1.Application, len(paths))
+		for _, p := range paths {
+			app, err := g.generateApplicationFromPath(appSet, p)
+			if err != nil {
+				log.Errorf("Error %e while  generating app from path: %s", err, p)
```

Comment:
If you'd like to take full advantage of structured logging I'd suggest:  log.WithError(err).WithField("path", p).Error("error while generating app from path")

Reply:
thanks!